File: backend/models/match.py
from sqlalchemy import Column, Integer, String, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from database import Base


class Match(Base):
    __tablename__ = "matches"

    id = Column(Integer, primary_key=True, index=True)
    team_home_id = Column(Integer, ForeignKey("teams.id"), nullable=False)
    team_away_id = Column(Integer, ForeignKey("teams.id"), nullable=False)
    date = Column(DateTime(timezone=True), nullable=False)
    stage = Column(String(20), nullable=False)
    score_home = Column(Integer)
    score_away = Column(Integer)
    status = Column(String(20), default="upcoming")

    team_home = relationship("Team", foreign_keys=[team_home_id])
    team_away = relationship("Team", foreign_keys=[team_away_id])

File: frontend/src/types/match.ts
export interface Team {
  id: number
  name: string
  code: string
  group: string
}

export interface Match {
  id: number
  date: string
  stage: string
  status: string | null
  score_home: number | null
  score_away: number | null
  team_home: Team
  team_away: Team
}
